### PR TITLE
TDP: Add additional document upload fields to support Spanish files

### PR DESCRIPTION
**Pull request by nbucknor**
- This PR Updates the ActivityPage model to allow for additional Teacher Guide and Student Handout files
- It also displays the "Activity Type: Available in Spanish" label on search results if that Activity has been tagged with the activity type "Available in Spanish". 

---

## Additions

- Classes ActivityPageActivityDocuments and ActivityPageHandoutDocuments
- Added "Activity Type" field to Search results only when an Activity is tagged "Available in Spanish"

## How to test this PR

1. Edit an Activity Page
2. You'll notice two new fields "Add Additional Activity Files" and "Add Additional Handout Files"
3. Both fields allow for the addition of multiple Documents to the Activity Page. 

## Screenshots
Screen Shot 2021-03-16 at 1 09 58 PM

Screen Shot 2021-03-16 at 1 15 31 PM

Screen Shot 2021-03-17 at 3 22 02 PM

## Notes and todos

- TODO: We could potentially migrate content from the following existing document fields into these  new fields:
  - Teacher guide
  - Student file 1
  - Student file 2
  - Student file 3

## Checklist

- [x] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [x] Future todos are captured in comments and/or tickets
- [ ] Project documentation has been updated, potentially one or more of:
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### CODE REVIEW

**Comment by nbucknor** [comment]
@Scotchester Is there a way I could see which **jenkins/functional-tests** are failing?

**Comment by Scotchester** [comment]
> @Scotchester Is there a way I could see which **jenkins/functional-tests** are failing?

@nbucknor Yeah, you have to replace the `dev-jenkins` hostname in that Details link with the real dev Jenkins domain. I'll connect with you privately to confirm you have that.

File: cfgov/teachers_digital_platform/models/activity_index_page.py [hunk 1]
```diff
@@ -349,6 +352,7 @@ def update_cards(self):
                 'summary': activity.summary,
                 'topic': activity.get_topics_list(),
                 'activity_duration': activity.activity_duration.title,
+                'available_in_spanish': 'Available in Spanish' if es else None,
```
Comment (Scotchester) [inline cfgov/teachers_digital_platform/models/activity_index_page.py:355]:
```suggestion
                'available_in_spanish': True if es else False,
```

If we pass this as a Boolean, then we can enter a translatable string in the template (`{{ _("Available in Spanish") }}`) which can itself be automatically translated into Spanish if a user has their browser's language set to Spanish, by including a `<span lang="es">` wrapper around the translated string. I think.

See https://cfpb.github.io/consumerfinance.gov/translation/ for more about how we do translations.

Even if the translation thing doesn't work, I think passing the boolean and having the string there in the template is a little cleaner.

**Review by Scotchester — COMMENTED** [review]
Haven't run it yet, but here are some initial comments.

One other broader one: I realize it would require a potentially-nasty data migration, but did you consider droping the individual `activity_file` and `handout_file` fields in favor of only using the new orderables?

**Comment by nbucknor** [comment]
> Haven't run it yet, but here are some initial comments.
> 
> One other broader one: I realize it would require a potentially-nasty data migration, but did you consider droping the individual `activity_file` and `handout_file` fields in favor of only using the new orderables?

Yeah. I had noted it as a potential TODO in this PR. I didn't consider it in scope of my given task and didn't want to spend 3x my estimate for this task working on an automated migration of the existing content.

File: cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_search_result.html [hunk 2]
```diff
@@ -28,6 +28,12 @@ <h5 class="activity_meta-key">Topic:</h5>
                 <h5 class="activity_meta-key">Activity duration:</h5>
                 {{ result.activity_duration }}
             </div>
+            {% if result.available_in_spanish %}
+            <div class="layout-row">
+                <h5 class="activity_meta-key">Activity type:</h5>
+                {{ result.available_in_spanish }}
+            </div>
+            {% endif %}
```
[the remark this answers is not in the document]
Comment (nbucknor) [reply cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_search_result.html:36]: > I'm not sure it's helpful to the user to show "Activity type:" here in the search result. Perhaps simply "Available in Spanish" (possibly bolded or highlighted some other way) would suffice?

I'll ask our UX Lead what she thinks. I agree with your assessment.

File: cfgov/teachers_digital_platform/models/activity_index_page.py (hunk 1 above)
Comment (nbucknor) [reply cfgov/teachers_digital_platform/models/activity_index_page.py]: Thanks, Scott. I'll refactor this.

**Comment by Scotchester** [comment]
> > Haven't run it yet, but here are some initial comments.
> > One other broader one: I realize it would require a potentially-nasty data migration, but did you consider droping the individual `activity_file` and `handout_file` fields in favor of only using the new orderables?
> 
> Yeah. I had noted it as a potential TODO in this PR. I didn't consider it in scope of my given task and didn't want to spend 3x my estimate for this task working on an automated migration of the existing content.

Oops, sorry, missed that TODO. Sounds good 👍

File: cfgov/teachers_digital_platform/models/activity_index_page.py (hunk 1 above)
Comment (Scotchester) [reply cfgov/teachers_digital_platform/models/activity_index_page.py]: Thanks for making that translatable. Could you please follow the steps on that doc page I linked to add the translated string to the list of available translations?

**Review by mrclay — APPROVED** [review]

File: cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_search_result.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (nbucknor) [reply cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_search_result.html]: @Scotchester  We ended up going with your suggestion and simply display "Available in Spanish" instead.

File: cfgov/teachers_digital_platform/models/activity_index_page.py (hunk 1 above)
Comment (Scotchester) [reply cfgov/teachers_digital_platform/models/activity_index_page.py]: > Thanks for making that translatable. Could you please follow the steps on [that doc page](https://cfpb.github.io/consumerfinance.gov/translation/) I linked to add the translated string to the list of available translations?

@nbucknor Re-upping this comment you may have missed.

Comment (nbucknor) [reply cfgov/teachers_digital_platform/models/activity_index_page.py]: @Scotchester I've run `manage.py makemessages -l es` and committed the updated django.po file. However, when I run `manage.py compilemessages` and commit the updated django.mo file, I get failed translation unit tests in travis CI. Am I missing a step?

Comment (Scotchester) [reply cfgov/teachers_digital_platform/models/activity_index_page.py]: @nbucknor Strange stuff! It appears that you reverted the committing of `django.mo` on this branch, correct? If I pull this branch down in its current state and run `compilemessages`, I don't get any changes on `django.mo`. So I guess this is good to go? 🤷‍♂️  I'll merge from main and see if things still pass.

**Review by Scotchester — APPROVED** [review]

**Pull request merged**